src/ReachSets/discretize.jl

```diff
@@ -1,310 +1,656 @@
-        Uset = inputset(cont_sys)
+       throw(ArgumentError("the exponentiation method $exp_method is unknown"))
+    end
+
+    return Φ₂_Aδ
+end
+
+"""
+    discretize_firstorder(𝑆, δ; [p], [exp_method])
+
+Apply a first-order approximation model to `S` obtaining a discrete initial
+value problem.
+
+### Input
+
+- `𝑆`           -- initial value problem for a continuous affine ODE with
+                   non-deterministic inputs
+- `δ`           -- step size
+- `p`           -- (optional, default: `Inf`) parameter in the considered norm
+- `exp_method`  -- (optional, default: `"base"`) the method used to take the matrix
+                    exponential of the coefficient matrix, choose among:
+
+    - `"base"` -- the scaling and squaring method implemented in Julia base,
+                  see `?exp` for details
+    - `"pade"` -- use Pade approximant method to compute matrix exponentials of
+                  sparse matrices, implemented in `Expokit`
+    - `"lazy"` -- compute a wrapper type around the matrix exponential, i.e. using
+                  the lazy implementation `SparseMatrixExp` from `LazySets` and
+                  the evaluation of the action of the matrix exponential using the
+                  `expmv` implementation from `Expokit`
+
+### Output
+
+The initial value problem for a discrete system. In particular:
+
+- if the input  system is homogeneous, a linear discrete system, `LinearDiscreteSystem`
+  is returned,
+- otherwise a constrained linear discrete systen is returned,
+  `ConstrainedLinearControlDiscreteSystem`.
+
+### Algorithm
+
+Let us define some notation. Let
+
+```math
+𝑆 : x' = Ax(t) + u(t)
+```
+with ``x(0) ∈ \\mathcal{X}_0``, ``u(t) ∈ U`` be the given continuous affine ODE
+`𝑆`, where `U` is the set of non-deterministic inputs and ``\\mathcal{X}_0``
+is the set of initial states.
+
+Define ``R_{\\mathcal{X}_0} = \\max_{x ∈ \\mathcal{X}_0} ‖x‖``,
+`D_{\\mathcal{X}_0} = \\max_{x, y ∈ \\mathcal{X}_0} ‖x-y‖`` and
+``R_{V} = \\max_{u ∈ U} ‖u‖``. If only the support functions of ``\\mathcal{X}_0``
+and ``V`` are known, these values might be hard to compute for some norms. See
+`Notes` below for details.
+
+Let ``Ω₀`` be the set defined as:
+```math
+Ω₀ = ConvexHull(\\mathcal{X}_0, e^{δA}\\mathcal{X}_0 ⊕ δU ⊕ αB_p)
+```
+where ``α = (e^{δ ‖A‖} - 1 - δ‖A‖)*(R_{\\mathcal{X}_0} + R_{U} / ‖A‖)`` and
+``B_p`` denotes the unit ball for the considered ``p``-norm.
+
+It is proved in [Lemma 1, 1] that the set of states reachable by ``S`` in the time
+interval ``[0, δ]``, that we denote ``R_{[0,δ]}(\\mathcal{X}_0)`` here,
+is included in ``Ω₀``:
+
+```math
+R_{[0,δ]}(\\mathcal{X}_0) ⊆ Ω₀.
+```
+
+Moreover, if ``d_H(A, B)`` denotes the Hausdorff distance between the sets ``A``
+and ``B`` in ``\\mathbb{R}^n``, then
+
+```math
+d_H(Ω₀, R_{[0,δ]}(\\mathcal{X}_0)) ≤ \\frac{1}{4}(e^{δ ‖A‖} - 1) D_{\\mathcal{X}_0} + 2α.
+```
+Hence, the approximation error can be made arbitrarily small by choosing ``δ``
+small enough.
```

Comment:
For the other algorithms you wrote this sentence, so I think it can be added here as well:
```Here we allow ``U`` to be a sequence of time varying non-deterministic input```